src/init.sh

```diff
@@ -170,5 +181,179 @@ function init_help()
+{
+  local command
+  local scope
+
+  command='git config'
+
+  if ! is_inside_work_tree; then
+    warning 'You are not in a git repository. All modifications will be global' 1>&2
+    if [[ $(ask_yN 'Would you like to proceed?' 'y') =~ '1' ]]; then
+      command+=' --global'
+    else
+      command=''
+    fi
+
+    printf '%s\n' "$command"
+    exit 0
+  fi
+
+  printf '\nSelect the scope of this configuration:\n' 1>&2
+
+  select scope in 'Local' 'Global'; do
+    case "$scope" in
+      'Global')
+        command+=' --global'
+        break
+        ;;
+      'Local')
+        command+=' --local'
+        break
+        ;;
+    esac
+  done
+
+  printf '%s\n' "$command"
+}
+
+# Allows the user to set the empty configurations if any.
+#
+# @_config_git: associative array with data about git configuration.
+# @flag: flag sended to the cmd_manager to change its behaviour.
+#
+# Returns: _config_git: array with the selected information configured on git
+function interactive_set_user_git_info()
+{
+  local -n _config_git="$1"
+  local flag="$2"
+  local git_editor_suggestion
+  local config_cmd
+  local scope
+
+  flag=${flag:-'SILENT'}
+
+  if [[ "${_config_git['configured']}" == '0' ]]; then
+    exit 0
+  fi
+
+  printf '\nNow Git will be configured.\n' 1>&2
+
+  config_cmd=$(set_git_config_scope)
+  if [[ -z "$config_cmd" ]]; then
+    return
+  fi
+
+  if [[ -z "${_config_git['name']}" ]]; then
+    _config_git['name']=$(ask_with_default 'What is your name?' "$USER")
+    cmd_manager "$flag" "$config_cmd user.name \"${_config_git['name']}\""
+  fi
+
+  if [[ -z "${_config_git['email']}" ]]; then
+    while :; do
+      _config_git['email']=$(ask_with_default 'What is your email?' '')
+
+      if validate_email "${_config_git['email']}"; then
+        break
+      fi
+
+      printf 'It was not possible to validate your e-mail, please try again.\n' 1>&2
+    done
+    cmd_manager "$flag" "$config_cmd user.email \"${_config_git['email']}\""
+  fi
+
+  if [[ -z "${_config_git['editor']}" ]]; then
+    if [[ $(ask_yN $'\nWould you like to configure your default editor on Git?' 'y') == '1' ]]; then
+
+      # This follows git precedence to determine used editor
+      git_editor_suggestion='vi'
+      [[ -n "$EDITOR" ]] && git_editor_suggestion="$EDITOR"
+      [[ -n "$VISUAL" ]] && git_editor_suggestion="$VISUAL"
+      _config_git['editor']=$(ask_with_default 'What is your main editor?' "$git_editor_suggestion")
+
+      cmd_manager "$flag" "$config_cmd core.editor \"${_config_git['editor']}\""
+    fi
+  fi
+
+  if [[ -z "${_config_git['branch']}" ]]; then
+    # This is a minor configuration, so default is "no"
+    if [[ $(ask_yN $'\nWould you like to configure your initial default branch on Git?' 'n') == '1' ]]; then
+      _config_git['branch']=$(ask_with_default 'What is your default branch name?' 'main')
+      cmd_manager "$flag" "$config_cmd init.defaultBranch \"${_config_git['branch']}\""
+    fi
+  fi
+}
```

Comment:
How about creating a function per question and calling them here? I think this will be easy to test and debug in the future.